// @smile-cdr/fhirts/dist/FHIR-R4/classes/researchElementDefinitionCharacteristic.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResearchElementDefinitionCharacteristic = void 0;
/**
 * The ResearchElementDefinition resource describes a \"PICO\" element that knowledge (evidence, assertion, recommendation) is about.
 */
class ResearchElementDefinitionCharacteristic {
}
exports.ResearchElementDefinitionCharacteristic = ResearchElementDefinitionCharacteristic;
(function (ResearchElementDefinitionCharacteristic) {
    ResearchElementDefinitionCharacteristic.StudyEffectiveGroupMeasureEnum = {
        Mean: 'mean',
        Median: 'median',
        MeanOfMean: 'mean-of-mean',
        MeanOfMedian: 'mean-of-median',
        MedianOfMean: 'median-of-mean',
        MedianOfMedian: 'median-of-median'
    };
    ResearchElementDefinitionCharacteristic.ParticipantEffectiveGroupMeasureEnum = {
        Mean: 'mean',
        Median: 'median',
        MeanOfMean: 'mean-of-mean',
        MeanOfMedian: 'mean-of-median',
        MedianOfMean: 'median-of-mean',
        MedianOfMedian: 'median-of-median'
    };
})(ResearchElementDefinitionCharacteristic = exports.ResearchElementDefinitionCharacteristic || (exports.ResearchElementDefinitionCharacteristic = {}));
